app/src/main/kotlin/com/backbase/golden_sample_app/presentation/MainActivity.kt

```diff
@@ -19,30 +25,23 @@ import org.koin.core.context.loadKoinModules
  */
 class MainActivity : AppCompatActivity() {
 
-    private lateinit var binding: ActivityMainBinding
+    lateinit var binding: ActivityMainBinding
+
     private val navigator: AppRouting by inject()
-    private lateinit var bottomNavigationHandler: BottomNavigationHandler
+    private val tabHeaderViewModel: TabHeaderViewModel by viewModels()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
-        val view = binding.root
-        setContentView(view)
-        bottomNavigationHandler = BottomNavigationHandler(
-            binding.bottomNavigation,
-            Navigation.findNavController(this, R.id.nav_host_container),
-            navigator
-        )
+        setContentView(binding.root)
 
-        if (savedInstanceState == null) {
-            bottomNavigationHandler.setupNavigation()
-            loadScopedDependencies()
-        }
-    }
+        val navController = findNavController(R.id.nav_host_container)
+        navigator.bind(navController)
+
+        updateStatusBarColor(isInRootScreen = tabHeaderViewModel.uiState.map { it.isInRootScreen })
```

Comment:
Status bar color in dark mode is also light blue

Reply:
Solved!